modules/gcp/cac-igm/cac-provisioning.sh.tmpl

```diff
@@ -156,13 +152,33 @@ wait_for_dc() {
     TIMEOUT=1200
     until netcat -vz ${domain_name} 636; do
         if [ $TIMEOUT -le 0 ]; then
+            log "--> Timed out trying to contact ${domain_name}:636. Continuing..."
             break
         else
-            log "Trying to contact ${domain_name}:636. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
+            log "--> Trying to contact ${domain_name}:636. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
         fi
         TIMEOUT=$((TIMEOUT-10))
         sleep 10
     done
+
+    # Wait for service account to be added. Do this last because it takes
+    # a while for new AD user to be added in a new Domain Controller.
+    # Note: using the domain controller IP instead of the domain name for
+    #       the host is more resilient.
+    log "--> Installing ldap_utils..."
+    RETRIES=5
+    while true; do
+        apt-get -qq update
+        apt-get -qq install ldap-utils
+        RC=$?
+        if [ $RC -eq 0 ] || [ $RETRIES -eq 0 ]; then
+            break
+        fi
+
+        log "--> ERROR: Failed to install ldap-utils. $RETRIES retries remaining..."
+        RETRIES=$((RETRIES-1))
+        sleep 5
+    done
```

Comment:
this probably won't work? ldapwhoami above requires ldap-utils , this section has to come first in wait_for_dc()